### PR TITLE
fix: Properly space Day One name

### PR DESCRIPTION
**Pull request by dcalhoun**
The Day One brand name includes a space between the two words.

<details><summary>Screenshots</summary>

| Before | After |
| - | - |
| day-one-before | day-one-after |

</details>

## To Test:

1. Check out the branch and apply the following testing diff:
    <details><summary>Testing Diff</summary>

    ```diff
    diff --git a/WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloggingprompts/BloggingPromptAttribution.kt b/WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloggingprompts/BloggingPromptAttribution.kt
    index 178c7da39b..d5947570ba 100644
    --- a/WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloggingprompts/BloggingPromptAttribution.kt
    +++ b/WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloggingprompts/BloggingPromptAttribution.kt
    @@ -36,11 +36,11 @@ enum class BloggingPromptAttribution(
              * for the BLOGANUARY campaign at this point, but might be in the future.
              */
             fun fromPrompt(
    -            prompt: BloggingPromptModel
    -        ): BloggingPromptAttribution = if (!prompt.bloganuaryId.isNullOrBlank()) {
    +            @Suppress("UNUSED_PARAMETER") prompt: BloggingPromptModel
    +        ): BloggingPromptAttribution = if (false) {
                 BLOGANUARY
             } else {
    -            fromString(prompt.attribution)
    +            fromString("dayone")
             }
         }
     }
    
    ```
    </details>
1. Build and launch the app.
1. Load the My Site tab and verify the "Day One" name in the Blogging Prompt attribution includes a space.

## Regression Notes

1. Potential unintended areas of impact
    Blogging Prompts, translations.
3. What I did to test those areas of impact (or what existing automated tests I relied on)
    Interacted with Blogging Prompts, test different device languages.
4. What automated tests I added (or what prevented me from doing so)
    Automated tests felt unnecessary for a typo.

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## UI Changes Testing Checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -2400,7 +2400,7 @@
     <string name="my_site_blogging_prompt_card_menu_view_more">View more prompts</string>
     <string name="my_site_blogging_prompt_card_menu_skip">Skip for today</string>
     <string name="my_site_blogging_prompt_card_menu_remove">Turn off prompts</string>
-    <string name="my_site_blogging_prompt_card_attribution_dayone">From &lt;b&gt;DayOne&lt;/b&gt;</string>
+    <string name="my_site_blogging_prompt_card_attribution_dayone">From &lt;b&gt;Day One&lt;/b&gt;</string>
```
Comment (dcalhoun) [inline WordPress/src/main/res/values/strings.xml]: It occurred to me that the `dayone` `name` might should be renamed to `day_one`, but I avoided doing so to limit the surface area of this change and avoid unintended regressions. I'm open to refactoring the `name` if it is deemed worthwhile.

Comment (dcalhoun) [reply WordPress/src/main/res/values/strings.xml]: Also, my understanding is that string files for other languages are automatically generated and I should not update them manually. Is that correct?

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/275">24.1</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (thomashorta) [reply WordPress/src/main/res/values/strings.xml]: So, we recently had a similar discussion and Olivier chimed in to help us understand better how stuff works on CI/localization side:

> In summary modifying the value of an existing key will cause a discrepancy in the release/* branch, between the codebase (which will expect the old copy as that's what it was when code was frozen) and the translations (updated with the new copy when we'll pull the latest from GlotPress)
> 
> TL;DR: prefer using a brand new key.
>
> https://github.com/wordpress-mobile/WordPress-Android/pull/19856#discussion_r1446710419

So I guess changing the `name` is actually desirable to avoid the problems mentioned by him there. Does that make sense?

> Also, my understanding is that string files for other languages are automatically generated and I should not update them manually. Is that correct?

That's correct! Even for a new string resource (new `name`) we should only touch this `main/res/strings.xml` file.

Comment (dcalhoun) [reply WordPress/src/main/res/values/strings.xml]: @thomashorta I refactored the changes to accommodate a new translation string. This is ready for another review.

**Review by thomashorta — APPROVED** [review]
The changes LGTM, thanks @dcalhoun !
